FreeRTOS/Test/CBMC/README.md

```diff
@@ -1,133 +1,133 @@
-  - Change into the top-level CBMC source directory and run
-    ```
-    cmake -H. -Bbuild -DCMAKE_BUILD_TYPE=Release -DWITH_JBMC=OFF
-    cmake --build build
-    ```
-
-- Ensure that you can run the programs `cbmc`, `goto-cc` (or `goto-cl`
-  on Windows), and `goto-instrument` from the command line. If you build
-  CBMC with CMake, the programs will have been installed under the
-  `build/bin/Debug` directory under the top-level `cbmc` directory; you
-  should add that directory to your `$PATH`. If you built CBMC using
-  Make, then those programs will have been installed in the `src/cbmc`,
-  `src/goto-cc`, and `src/goto-instrument` directories respectively.
-
-
-### Setting up the proofs
-
-Change into the `proofs` directory. On Windows, run
-```
-python prepare.py
-```
-On macOS or Linux, run
-```
-./prepare.py
-```
-If you are on a Windows machine but want to generate Linux Makefiles (or vice
-versa), you can pass the `--system linux` or `--system windows` options to those
-programs.
-
-
-### Running the proofs
-
-Each of the leaf directories under `proofs` is a proof of the memory
-safety of a single entry point in a:FR. The scripts that you ran in the
-previous step will have left a Makefile in each of those directories. To
-run a proof, change into the directory for that proof and run `nmake` on
-Windows or `make` on Linux or macOS. The proofs may take some time to
-run; they eventually write their output to `cbmc.txt`, which should have
-the text `VERIFICATION SUCCESSFUL` at the end.
-
-
-### Proof directory structure
-
-This directory contains the following subdirectories:
-
-- `proofs` contains the proofs run against each pull request
-- `patches` contains a set of patches that get applied to the codebase prior to
-  running the proofs
-- `include` and `windows` contain header files used by the proofs.
+CBMC Proof Infrastructure
+=========================
+
+This directory contains automated proofs of the memory safety of various parts
+of the amazon:FreeRTOS codebase. A continuous integration system validates every
+pull request posted to the repository against these proofs, and developers can
+also run the proofs on their local machines.
+
+The proofs are checked using the
+[C Bounded Model Checker](http://www.cprover.org/cbmc/), an open-source static
+analysis tool
+([GitHub repository](https://github.com/diffblue/cbmc)). This README describes
+how to run the proofs on your local clone of a:FR.
+
+
+Bulding and running proofs
+--------------------------
+
+For historical reasons, some of the proofs are built and run using CMake
+and CTest. Others use a custom python-based build system. New proofs
+should use CMake. This README describes how to build and run both kinds
+of proof.
+
+
+CMake-based build
+-----------------
+
+Follow the CBMC installation instructions below.
+
+Suppose that the amazon-freertos source tree is located at
+`~/src/amazon-freertos` and you wish to build the proofs into

```

Comment:
Need to change this to FreeRTOS paths and commands

Reply:
Done